wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -267,78 +194,58 @@ def wait_job_launch(
-
-        batch_api = kubernetes.client.BatchV1Api(api_client)
-        core_api = kubernetes.client.CoreV1Api(api_client)
-
-        # allow users to specify template or entire spec
-        if resource_args.get("job_spec"):
-            job_dict = load_json_yaml_dict(resource_args["job_spec"])
-        else:
-            # begin constructing job sped
-            job_dict = {"apiVersion": "batch/v1", "kind": "Job"}
-
-        # extract job spec component parts for convenience
-        job_metadata = job_dict.get("metadata", {})
-        job_spec = job_dict.get("spec", {})
-        pod_template = job_spec.get("template", {})
-        pod_metadata = pod_template.get("metadata", {})
-        pod_spec = pod_template.get("spec", {})
-        containers = pod_spec.get("containers", [{}])
-        job_status = job_dict.get("status", {})
-
-        # begin pulling resource arg overrides. all of these are optional
+        namespace: str,
+        core_api: "CoreV1Api",
+    ) -> Tuple[Dict[str, Any], Optional["V1Secret"]]:
+        """Apply our default values, return job dict and secret."""
+        job: Dict[str, Any] = {
+            "apiVersion": "batch/v1",
+            "kind": "Job",
+        }
+        job.update(resource_args)
 
-        # allow top-level namespace override, otherwise take namespace specified at the job level, or default in current context
-        default_namespace = (
-            context["context"].get("namespace", "default") if context else "default"
-        )
-        namespace = self.get_namespace(resource_args) or default_namespace
+        job_metadata: Dict[str, Any] = job.get("metadata", {})
+        job_spec: Dict[str, Any] = {"backoffLimit": 0, "ttlSecondsAfterFinished": 60}
+        job_spec.update(job.get("spec", {}))
+        pod_template: Dict[str, Any] = job_spec.get("template", {})
+        pod_spec: Dict[str, Any] = {"restartPolicy": "Never"}
+        pod_spec.update(pod_template.get("spec", {}))
+        containers: List[Dict[str, Any]] = pod_spec.get("containers", [{}])
 
-        # name precedence: resource args override > name in spec file > generated name
-        job_metadata["name"] = resource_args.get("job_name", job_metadata.get("name"))
+        # name precedence: name in spec > generated name
         if not job_metadata.get("name"):
             job_metadata["generateName"] = make_name_dns_safe(
                 f"launch-{launch_project.target_entity}-{launch_project.target_project}-"
             )
 
-        if resource_args.get("job_labels"):
-            job_metadata["labels"] = resource_args.get("job_labels")
-
-        self.populate_job_spec(job_spec, resource_args)
-        self.populate_pod_spec(pod_spec, resource_args)
-
-        self.populate_container_resources(containers, resource_args)
-
-        # cmd
-        entry_point = launch_project.get_single_entry_point()
+        for i, cont in enumerate(containers):
```

Comment:
Should this code block go into defaults? And should users be able to override them?